### PR TITLE
feature: update TFlint #2418

### LINKED ISSUE #2410: [Bug] tflint Has Module Inspection Enabled by Default
**Summary**

The Terragrunt `tflint` integration has module inspection enabled by default, set in [tflint.go:53](https://github.com/gruntwork-io/terragrunt/blob/master/tflint/tflint.go#L53) via  "--module". 

I am not able to deactivate it, even with my config saying otherwise.

In tflint itself, this option is disabled by default (see [here](https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/module-inspection.md#enabling)), meaning one has to set it explicitly in the configuration or via CLI.

**Expected Behavior**
Module inspection is disabled by default, enabling is an explicit task

### PR DESCRIPTION
**Pull request by theurichde**
## Description
Updates TFlint to v0.44.1

Fixes #2410 #2418 #2420 

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Updated TFlint to v0.44.1
Removed forced `--module` check for TFlint

### Migration Guide

If you want to have the TFlint module check enabled, simply add `module = true` to your `.tflint.config.hcl`

```
config {
  module = true
}
```

### CODE REVIEW

**Comment by denis256** [comment]
Hi,
I'm not sure if this PR is ready, but it fails now to build because of issues with `go.sum`

```
7 errors occurred:
--> linux/amd64 error: exit status 1
Stderr: ../go/pkg/mod/github.com/hashicorp/terraform@v0.15.3/lang/funcs/crypto.go:21:2: missing go.sum entry for module providing package golang.org/x/crypto/bcrypt (imported by github.com/hashicorp/terraform/lang/funcs); to add:
	go get github.com/hashicorp/terraform/lang/funcs@v0.15.3
../go/pkg/mod/filippo.io/age@v1.0.0-beta7/internal/stream/stream.go:15:2: missing go.sum entry for module providing package golang.org/x/crypto/chacha20poly1305 (imported by filippo.io/age); to add:
	go get filippo.io/age@v1.0.0-beta7
../go/pkg/mod/filippo.io/age@v1.0.0-beta7/x25519.go:20:2: missing go.sum entry for module providing package golang.org/x/crypto/curve25519 (imported by filippo.io/age); to add:
	go get filippo.io/age@v1.0.0-beta7
../go/pkg/mod/gopkg.in/square/go-jose.v2@v2.5.1/asymmetric.go:31:2: missing go.sum entry for module providing package golang.org/x/crypto/ed25519 (imported by gopkg.in/square/go-jose.v2); to add:
	go get gopkg.in/square/go-jose.v2@v2.5.1
../go/pkg/mod/filippo.io/age@v1.0.0-beta7/primitives.go:17:2: missing go.sum entry for module providing package golang.org/x/crypto/hkdf (imported by filippo.io/age); to add:
	go get filippo.io/age@v1.0.0-beta7
../go/pkg/mod/go.mozilla.org/sops/v3@v3.7.2/pgp/keysource.go:25:2: missing go.sum entry for module providing package golang.org/x/crypto/openpgp (imported by github.com/terraform-linters/tflint/plugin); to add:
	go get github.com/terraform-linters/tflint/plugin@v0.44.1
../go/pkg/mod/go.mozilla.org/sops/v3@v3.7.2/pgp/keysource.go:26:2: missing go.sum entry for module providing package golang.org/x/crypto/openpgp/armor (imported by go.mozilla.org/sops/v3/pgp); to add:
	go get go.mozilla.org/sops/v3/pgp@v3.7.2
../go/pkg/mod/gopkg.in/square/go-jose.v2@v2.5.1/symmetric.go:33:2: missing go.sum entry for module providing package golang.org/x/crypto/pbkdf2 (imported by gopkg.in/square/go-jose.v2); to add:
	go get gopkg.in/square/go-jose.v2@v2.5.1
../go/pkg/mod/github.com/!azure/go-autorest/autorest/adal@v0.9.13/persist.go:27:2: missing go.sum entry for module providing package golang.org/x/crypto/pkcs12 (imported by github.com/Azure/go-autorest/autorest/adal); to add:
	go get github.com/Azure/go-autorest/autorest/adal@v0.9.13
../go/pkg/mod/filippo.io/age@v1.0.0-beta7/internal/stream/stream.go:16:2: missing go.sum entry for module providing package golang.org/x/crypto/poly1305 (imported by filippo.io/age/internal/stream); to add:
	go get filippo.io/age/internal/stream@v1.0.0-beta7
../go/pkg/mod/filippo.io/age@v1.0.0-beta7/scrypt.go:17:2: missing go.sum entry for module providing package golang.org/x/crypto/scrypt (imported by filippo.io/age); to add:
	go get filippo.io/age@v1.0.0-beta7
../go/pkg/mod/github.com/hashicorp/terraform@v0.15.3/lang/funcs/crypto.go:22:2: missing go.sum entry for module providing package golang.org/x/crypto/ssh (imported by github.com/hashicorp/terraform/lang/funcs); to add:

```

**Comment by theurichde** [comment]
Hi @denis256. Thanks for the hint! I ran a god mod tidy and pushed the clean go.sum - let's see how it goes. 

Guess I resolved the merge conflict wrong yesterday :-/

Besides that, the PR is ready from my side  - I will mark it as ready if the tests pass successfully.

**Comment by theurichde** [comment]
I am fixing the failing test. Deleted too much content in the tf files. -.-'

**Comment by denis256** [comment]
```
--- PASS: TestTflintFindsNoIssuesWithValidCode (2.33s)
=== RUN   TestTflintFindsModule
    integration_test.go:3882: Copying fixture-tflint/module-found to /tmp/terragrunt-test3238892953
runTerragruntVersionCommand after split
[terragrunt plan --terragrunt-working-dir /tmp/terragrunt-test3238892953/fixture-tflint/module-found]
    integration_test.go:5254: 
        	Error Trace:	integration_test.go:5254
        	Error:      	Received unexpected error:
        	            	1 error occurred:
        	            		* Tflint found issues in the project. Check for the tflint logs.
        	            	
        	Test:       	TestTflintFindsModule
--- FAIL: TestTflintFindsModule (4.88s)
```

File: test/fixture-tflint/module-found/terragrunt.hcl [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  source = "."
+
+  before_hook "tflint" {
+    commands = ["apply", "plan"]
+    execute = ["tflint"]
+  }
+}
+
+inputs = {
+  aws_region = "eu-central-1"
```
[the remark this answers is not in the document]
Comment (denis256) [reply test/fixture-tflint/module-found/terragrunt.hcl:11]: There is a constant `TERRAFORM_REMOTE_STATE_S3_REGION` with used region

**Comment by denis256** [comment]
```
[INFO] Initializing environment for https://github.com/gruntwork-io/pre-commit.
Terraform fmt............................................................Failed
- hook id: terraform-fmt
- files were modified by this hook

main.tf
```

**Comment by theurichde** [comment]
`build and test`is green 🥳

**Review by marinalimeira — APPROVED** [review]
Thank you for reporting the bug and fixing it!

The code changes looks good to me. Thanks @denis256 for the first round of review :) 

I will merge and release this, also I will take care of updating the docs website, to mention the `module = true`.

**Comment by marinalimeira** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.43.0

**Pull request merged**